src/main/webapp/script.js

```diff
@@ -312,38 +313,44 @@ function buildInput(type, value, className, name) {
   return form;
 }
 
-// Get all location input fields, and add the Google Places autofill.
-function addLocationAutofill() {
-  let locationInputs = document.getElementsByClassName('places-autofill');
-  Array.prototype.forEach.call(locationInputs, (locationInput) => {
-    let locationAutocomplete = new google.maps.places.Autocomplete(locationInput);
-
-    // Any time the input changes through user typing, remove the 'is-valid' class.
-    // This will not be called if the user clicks on Google Place autofill.
-    locationInput.addEventListener('input', () => {
-      locationInput.classList.remove('is-valid');
-
-      // For "input destination" field, check "Next" button; for POI, check 
-      // "Add POI" button.
-      if (locationInput.id === 'inputDestination') {
-        checkNextButton();
-      } else if (locationInput.id === 'inputPoi') {
-        checkAddPoiButton();
-      }
-    });
+// Add text input POI Google Places autofill.
+function addInputPoiLocationAutofill() {
```

Comment:
You could parameterize this function probably to make it work for both inputPoi and inputDestination, but I'm not overly fussed. If you do decide to parameterize though, ask for me to re-review